### ISSUE TITLE
add save/download button to report

### ISSUE THREAD

**Issue by brendankenny** [GoogleChrome/lighthouse#1054]
Saving the page should just work since all assets are inlined, but the browser's "Save Page As" has always been awkward and error prone, so users avoid it. Should be easy to instead add a save (or download) link to the report.

Something like `<a href="#" download="result.html">save me, please</a>` looks like it works correctly with the blobURL-based report (after #908), and for nice symmetry it works with the CLI generated report as well, though of course you already have the CLI report in your file system if you're looking at it.

Could also customize the `download` attribute's value to match the [CLI report naming](https://github.com/GoogleChrome/lighthouse/blob/2891de98629cf6a0787052a140da831f0cc8dd85/lighthouse-core/lib/asset-saver.js#L23-L35) so you don't get collisions when saving multiple reports.

Need a nice save link or icon. Maybe replace the print icon?

@paulirish since I believe you were thinking about this and sharing reports after doing #908

**Comment by brendankenny** [comment]
thinking about sharing, some kind of data URI could work, but the reports are very big, so it's probably not workable. It would be nice to have a [timeline-viewer](https://chromedevtools.github.io/timeline-viewer/)-like solution for sharing, though

**Comment by ebidel** [comment]
I can take this. The file apis and me are buddies.

**Comment by paulirish** [comment]
Agree this should be super easy with the blob url. like 5 lines.

Chris had a few download options in his mock. Download as HTML, JSON or PDF. 

I think we can offer "download HTML", "download JSON" or "print" quite easily.

**Comment by ebidel** [comment]
Ref:

<img width="307" alt="screen shot 2016-11-28 at 3 52 14 pm" src="https://cloud.githubusercontent.com/assets/238208/20690975/a5602924-b582-11e6-9276-5c83ded9f0e2.png">

**Comment by brendankenny** [comment]
solved in #1182 for the viewer, but re-opening until we figure out the great report/viewer consolidation for report support :)

**Comment by ebidel** [comment]
Closing this in favor of https://github.com/GoogleChrome/lighthouse/issues/1166
